### ISSUE TITLE
1.13 nbt.region support

### ISSUE THREAD

**Issue by jason-green-io** [twoolie/NBT#95]
I think either the mca file format or the chunk format changed, any chance we can get nbt working again with 1.13?

**Comment by suresttexas00** [comment]
1.13 of what?

**Comment by jason-green-io** [comment]
1.13 of Minecraft

**Comment by macfreek** [comment]
Thanks for reporting this.

I'm currently not actively playing Minecraft, so development on NBT is slow. That said, I think this should take priority. If you have an example file and/or pointers to the changes in the chunk format, that would be appreciated.

**Comment by jason-green-io** [comment]
Here's some details on the changes: https://www.minecraftforum.net/forums/minecraft-java-edition/recent-updates-and-snapshots/2894808-minecraft-1-13-new-chunk-format-fully-decoded-read

I'll upload a r.0.0.mca a bit later.

**Comment by jason-green-io** [comment]
[r.0.0.mca.gz](https://github.com/twoolie/NBT/files/1998993/r.0.0.mca.gz)

**Comment by macfreek** [comment]
Thanks @jason-green-io. I just pushed a branch dev-long-array based on on https://minecraft.gamepedia.com/NBT_format.

Basically, it seem the new part is an array with longs (32-bit integers).
Since I have not tested this (and I don't know if these 32-bit integers are big endian or little endian, I am not confident yet to push it to master.

**Comment by jason-green-io** [comment]
I'll give it a go

**Comment by jason-green-io** [comment]
I think you forgot to add the tag to TAGLIST :)

**Comment by macfreek** [comment]
Yup, but it seems that the Long array is actually just a minor thing from last year which wasn't used in practice. The 1.13 NBT change seems unrelated and much deeper.

**Comment by jason-green-io** [comment]
Yup, I'm still getting errors on parsing regions.

**Comment by macfreek** [comment]
Unfortunately, it is getting late in my time zone, and will be busy with work coming week. I'll try to look later, but all suggestions and code contributions are welcome.

**Comment by jason-green-io** [comment]
Cool, I'll see if I can get my head around the region format.

**Comment by jason-green-io** [comment]
Here some source code for a tool that can read the new region format. 

'''
Minecraft Tools/Substrate_Jupisoft/AnvilSection.cs
'''

That is probably what you want

**Comment by macfreek** [comment]
I spotted my error: I defined type type of NBT `long` as a [Struct](https://docs.python.org/3/library/struct.html#format-characters) `long`. That's wrong. a NBT long is 8 bytes, a struct `long` is 4 bytes. I needed to pick the struct `long long`.

Unfortunately, some more fixes need to be made. I'll see if I can upload my test scripts too. I'm a bit hesitant to put them in the examples because they are not very common. 

Looking at the examples directory mostly makes it clear to me that NBT still need the overhaul I once planned. It took me 28 lines of code to reliably open both compressed and uncompressed .dat (NBT) files.

**Comment by jason-green-io** [comment]
That works! Thanks!
